src/Ecs/Systems/DamageModifierSystem.gd

```diff
@@ -0,0 +1,14 @@
+extends "res://Ecs/Core/System.gd"
+
+var _event_bus
+
+func _init(event_bus):
+	event_bus.connect('damage_entity', self, 'on_damage_entity')
+	_event_bus = event_bus
+
+func on_damage_entity(entity, damage):
+    if entity.has("StatusEffectsComponent"):
+        var effects_component = entity.get("StatusEffectsComponent")
+        for effect in effects_component.effects:
+            damage = effect.process_damage(damage)
```

Comment:
This is very interesting - almost like "filter design pattern." I wonder if/when the order of statuses inflicted will matter?

Reply:
> I wonder if/when the order of statuses inflicted will matter?

It will matter when we add statuses with non-commutative effects, and that means not right now. We can worry about that later, if/when we need to.